designs/Prerelease-Option.md

```diff
@@ -0,0 +1,69 @@
+# Spec Name
+
+* Status: In Review
+* Author(s): [Martin Ruiz](https://github.com/martinrrm)
+* Issue: [4699](https://github.com/NuGet/Home/issues/4699) dotnet add package - support for NuGet "Pre" version
+
+## Problem Background
+
+Customers want to add the latest version of a package to their project, even if the latest version is prerelease.
+
+## Who are the customers
+
+All dotnet.exe customers
+
+## Goals
+
+Add an option to the command `add package` to enable installing latest prerelease version.
+
+## Non-Goals
+
+## Solution
+
+Add the option `--prerelease` in the command `add package`. This command will get the latest version of the package available even if it is not a stable version.
```

Comment:
Has there been any consideration to allow specification of prerelease-enabled floating versions with the `dotnet add package` command? for example:

`dotnet add package newtonsoft.json *-*`

Today `dotnet add package newtonsoft.json 12.*` will work, but `dotnet add package newtonsoft.json *-*` will return `error: '*-*' is not a valid version string.`

Reply:
Maybe you would need to add `--prerelease` so that would work?:
`dotnet add package --prerelease newtonsoft.json *-*`

Reply:
`dotnet add package Nuget.common --version "*-*"` works for me. 

Remember you need 3.1.300 for `*-*` support.

Reply:
OOOHHHH that makes sense, thanks 👌